pkg/summarizer/common/common.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2020 The TestGrid Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+const (
+	// HighlyFlaky is the flakiness threshold for tests of flakiness [20-100)
+	HighlyFlaky = 20.0
+	// SomewhatFlaky is the flakiness threshold for tests of flakiness [10-20)
+	SomewhatFlaky = 10.0
+	// SlightlyFlaky is the flakiness for tests of flakiness [3-10)
+	SlightlyFlaky = 3.0
+	// NonFlaky is the flakiness threshold for tests of flakiness [0-3)
+	NonFlaky = 0.0
+)
+
+// GridMetrics contains the gathered metrics such as passed and failed test count
+// for a state.proto Grid
+type GridMetrics struct {
+	Name             string
+	Passed           int
+	Failed           int
+	FlakyCount       int
+	AverageFlakiness float64
+	FailedInfraCount int
+	InfraFailures    map[string]int
+}
+
+// NewGridMetrics constructs a new GridMetrics struct with nil default values
+// reassigned to working versions.
+func NewGridMetrics(name string) *GridMetrics {
+	var gridMetrics GridMetrics
```

Comment:
nit: I think this can become

`gridMetrics := GridMetrics{Name: name, InfraFailures: make(map[string]int, 0)}`

Reply:
Done